### PR TITLE
 prepare blog section

### PR DESCRIPTION
**Pull request by houssembaltii**
## `Blog` Schema

- **`title`**: `string`  
  The title of the blog post.
- **`description`**: `string` (optional)  
  A brief summary or preview text for the blog post.

- **`BlogImage`**: `object` (optional)  
  An optional image object with:
  - **`src`**: `string`  
    The URL of the image.
  - **`alt`**: `string`  
    Alternative text for the image.

- **`author`**: `object`  
  Details about the author, including:
  - **`name`**: `string`  
    The name of the author.
  - **`Linkedin`**: `string` (optional)  
    The author's LinkedIn profile URL.

- **`date`**: `date`  
  The publication date of the blog post. 

- **`readingTime`**: `number` (optional)  
  The estimated reading time of the blog post, in minutes.

- **`tags`**: `array` of `string` (optional)  
  Tags related to the blog post.

- **`isTopPost`**: `boolean`  
  A flag indicating whether the blog post should be featured prominently on the homepage.

### CODE REVIEW

File: src/content/blogs/blogs.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    title: z.string(),
+    // Preview text
+    description: z.string().optional(),
+    BlogImage: z
+      .object({
+        src: z.string(),
+        alt: z.string(),
+      })
+      .optional(),
+    author: z.object({
```
Comment (JeanneGrenet) [inline src/content/blogs/blogs.ts:15]: I think we can add an optional picture for the author because we already have pictures for speakers and organizers

**Review by JeanneGrenet — CHANGES_REQUESTED** [review]

File: src/content/blogs/blogs.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { z, defineCollection } from "astro:content";
+
+export type Blog = z.infer<ReturnType<typeof zBlog>>;
+const zBlog = () =>
+  z.object({
+    title: z.string(),
+    // Preview text
+    description: z.string().optional(),
+    BlogImage: z
```
Comment (yoannfleurydev) [inline src/content/blogs/blogs.ts:9]:
```suggestion
    featuredImage: z
```

File: src/content/blogs/blogs.ts (hunk 1 above)
Comment (yoannfleurydev) [reply src/content/blogs/blogs.ts:15]: We will need to reference a `people` collection. We do not need to redefine people in every collection then

File: src/content/blogs/blogs.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        alt: z.string(),
+      })
+      .optional(),
+    author: z.object({
+      name: z.string(),
+      Linkedin: z.string().optional(),
+      image: z.string().optional(),
+    }),
+    date: z.date(),
+    readingTime: z.number().optional(),
```
Comment (yoannfleurydev) [inline src/content/blogs/blogs.ts:21]: Can be computed, we don't want to bother providing this value.

File: src/content/blogs/blogs.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    tags: z.array(z.string()).optional(),
+    //  This attribute is used to select certain blogs to be featured on the homepage
+    isTopPost: z.boolean(),
```
Comment (yoannfleurydev) [inline src/content/blogs/blogs.ts:24]: Can be done in a config file, using blog post slug, so blog post don't hold this information

**Review by yoannfleurydev — CHANGES_REQUESTED** [review]

**Review by yoannfleurydev — APPROVED** [review]

**Pull request merged**